xarray/core/variable.py

```diff
@@ -123,6 +123,11 @@ def as_variable(
     if isinstance(obj, Variable):
         obj = obj.copy(deep=False)
     elif isinstance(obj, tuple):
+        try:
+            dims_, data_, *attrs = obj
```

Comment:
Here `obj` has been expanded to the correct form. Can't we use this downstream as well? 
The use of `obj[1]` and `*obj` a little below doesn't seem necessary anymore, and maybe even the try/except is not necessary anymore either.

Reply:
I assume that the code can be written like that:

- Remove the obj[1] in favor of data
- Pass instead of Variable(*obj) -> Variable(dims_, data_, *attrs)

Although I would not remove the try block just in case something comes up on the conversion while the tuple is correct.

Reply:
I'm thinking about this try/except a little further down:
```python
        try:
            obj = Variable(*obj)
        except (TypeError, ValueError) as error:
            raise error.__class__(
                f"Variable {name!r}: Could not convert tuple of form "
                f"(dims, data[, attrs, encoding]): {obj} to Variable."
)
```
It is the same error you have added, just a little too late. It's not needed anymore with your addition as I think Variable will handle argument checking better than this function.

Reply:
Well I have added an extraction check to see if the tuple meets all the condition, by removing it, we will remove the ability to try to create a variable from the tuple. Of course, I assume that if the tuple is incomplete this cannot be done in any case so it does not make sense and that is how it overlaps with the check i added.